settings.gradle

```diff
@@ -13,35 +13,79 @@ if ( !JavaVersion.current().java8Compatible ) {
     throw new GradleException( "Gradle must be run with Java 8 or later" )
 }
 
-// Consume the property 'testedJavaVersion' here and
-// set it on gradle.ext so that we can inspect the result in build.gradle.
-// We wouldn't be able to do that if we consumed
-// the property in base-information.gradle and set it on project.ext.
-
-if ( hasProperty( 'testedJavaVersion' ) ) {
-    logger.warn( "[WARN] Targeting Java version '$testedJavaVersion' in tests." )
-    gradle.ext.testedJavaVersion = testedJavaVersion
-    gradle.ext.testedJavaVersionAsEnum = JavaVersion.toVersion( testedJavaVersion )
-}
-else {
-    // We will simply use Gradle's JDK for compilation, tests and javadoc generation.
-    def major
-    if ( JavaVersion.current() == JavaVersion.VERSION_HIGHER) {
-        logger.warn( "Gradle does not support this JDK, because it is too recent; build is likely to fail." +
-                " To avoid failures, you should specify an older Java version in the 'testedJavaVersion' parameter." +
-                " Just append the following to your gradle command:" +
-                " '-PtestedJavaVersion=<major version of your newer JDK, e.g. 14>'" )
-        // Use a hack to retrieve the major as a string.
-        // This only works for Java 9+ (we're at least on Java 18 here).
-        gradle.ext.testedJavaVersion = System.getProperty( 'java.specification.version' )
+gradle.ext.baselineJavaVersion = JavaLanguageVersion.of( 8 )
+
+// Gradle does bytecode transformation on tests.
```

Comment:
hu?? Have a pointer about what kinds of things it needs doing?

Reply:
Not really, but you can try setting the max supported version to 16 and running the build on 16: you'll see errors related to "incompatible minor.major" pop up, originating from Gradle.

Maybe Gradle just inspects the bytecode, I don't know. But it definitely is sensitive to the bytecode version.

Reply:
ok, thanks.